app/api/transportations_api.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class TransportationsApi < Grape::API
+  format :json
+
+  helpers AuthorizationHelper, CarbonFootprintHelper
+
+  namespace 'transportation' do
+    # POST /transportations
+    desc 'Create transportation' do
+      tags %w[transportation]
+      http_codes [
+        { code: 201, message: 'Create transportation' },
+        { code: 400, message: 'Bad request!' }
+      ]
+    end
+    desc 'Headers', {
+      headers: {
+        'Auth-Token' => {
+          description: 'Validates your identity',
+          optional: true
+        }
+      }
+    }
+
+    params do
+      optional :flights, type: Array[JSON] do
+        requires :from, type: String, desc: 'from', documentation: { param_type: 'body' }
+        requires :to, type: String, desc: 'to', documentation: { param_type: 'body' }
+      end
+      optional :public_transports, type: Array[JSON] do
+        requires :transport_type, type: Integer, desc: 'transport_type', documentation: { param_type: 'body' }
+        requires :total_km, type: Float, desc: 'km', documentation: { param_type: 'body' }
+      end
+      optional :cars, type: Array[JSON] do
+        requires :fuel_type, type: Integer, desc: 'fuel_type', documentation: { param_type: 'body' }
+        requires :fuel_consumption, type: Float, desc: 'fuel_consumption', documentation: { param_type: 'body' }
+        requires :total_km, type: Float, desc: 'total_km', documentation: { param_type: 'body' }
+      end
+    end
+    post do
+      cars_params = params[:cars]
+      flights_params = params[:flights]
+      public_transports_params = params[:public_transports]
+      token = headers.fetch('Auth-Token', nil)
+      if token
+        user = authorize_user(token)
+        footprint = Footprint.new(user_id: user.id)
+      else
+        footprint = Footprint.new
+      end
+      error!(footprint.errors, 400) unless footprint.present? && footprint.save
+
+      # TODO: API CALL TO FIND THE FLIGHT DISTANCE BETWEEN AIRPORTS
```

Comment:
We might also move this call to a background job in order to avoid the timeout for our call.